fluent-langneg/src/matches.js

```diff
@@ -0,0 +1,210 @@
+/* eslint no-magic-numbers: 0 */
+/* eslint complexity: ["error", { "max": 22 }] */
+
+import Locale from './locale';
+import { getLikelySubtagsMin } from './subtags';
+
+/**
+ * Attempts to build a most likely maximum version of the locale id based
+ * on it's minimal version.
+ *
+ * If possible it uses the lookup table above. If this one is missing,
+ * it only attempts to place the language code in the region position.
+ */
+function getLikelySubtagsLocale(loc, likelySubtags) {
+  if (likelySubtags) {
+    for (const key of Object.keys(likelySubtags)) {
+      if (key.toLowerCase() === loc.toLowerCase()) {
+        return new Locale(likelySubtags[key]);
+      }
+    }
+    return null;
+  }
+  return getLikelySubtagsMin(loc);
+}
+
+/**
+ * Replaces the region position with a range to allow for matches
+ * with the same language/script but from different region.
+ */
+function regionRangeFor(locale) {
+  locale.region = '*';
+  return locale;
+}
+
+function variantRangeFor(locale) {
+  locale.variant = '*';
+  return locale;
+}
```

Comment:
I'm working on the C++ port right now for Gecko.  Once I'm done with it, I'll update the JS implementation to match it. It does have this change as well.